lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,180 @@
+ * @param {string} sourceCode - sourceCode of the Object.assign call
+ * @returns {array} formatted args - replaces the Object.assign with a spread object.
+ */
+function parseArgs(args, sourceCode) {
+    const mapped = args.map((arg, i) => {
+
+        // If the the argument is an empty object
+        if (arg.type === "ObjectExpression" && arg.properties.length === 0) {
+            return "";
+        }
+
+        // if the argument is another object.assign call run this function for all of it's arguments
+        if (isObjectAssign(arg)) {
+            return parseArgs(arg.arguments, sourceCode);
+        }
+
+        const next = args[i + 1] || {};
+
+        // if the argument is an object with properties
+        if (arg.type === "ObjectExpression") {
+            const trimmedObject = stripCurlies(sourceCode.getText(arg)).trim();
+
+            return next.range && next.range[0] ? `${trimmedObject}, ` : trimmedObject;
+        }
+
+        return next.range && next.range[0]
+            ? `...${sourceCode.getText(arg)}, `
+            : `...${sourceCode.getText(arg)}`;
+    });
+
+    return [].concat.apply([], mapped);
+}
+
+/**
+ * Autofixes the Object.assign call to use an object spread instead.
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @param {string} sourceCode - sourceCode of the Object.assign call
+ * @returns {Function} autofixer - replaces the Object.assign with a spread object.
+ */
+function autofixSpread(node, sourceCode) {
+    return fixer => {
+        const args = node.arguments;
+
+        const processedArgs = parseArgs(args, sourceCode).join("");
+
+        const lastArg = tail(args);
+        const firstArg = args[0];
+
+        const funcEnd = sourceCode.text
+            .slice(lastArg.range[1], node.range[1])
+            .split(")")[0];
+        const funcStart = tail(
+            sourceCode.text.slice(node.range[0], firstArg.range[0]).split("(")
+        );
+
+        return fixer.replaceText(
+            node,
+            `{${funcStart}${processedArgs}${funcEnd}}`
+        );
+    };
+}
+
+/**
+ * Autofixes the Object.assign call with a single object literal as an argument
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @param {string} sourceCode - sourceCode of the Object.assign call
+ * @returns {Function} autofixer - replaces the Object.assign with a object literal.
+ */
+function autofixObjectLiteral(node, sourceCode) {
+    return fixer => {
+        const argument = node.arguments[0];
+
+        return fixer.replaceText(node, sourceCode.text.slice(argument.range[0], argument.range[1]));
+    };
+}
+
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "disallow using Object.assign with an object literal as the first argument and prefer the use of object spread instead.",
+            category: "ECMAScript 6",
```

Comment:
Would you change the category to `Stylistic Issues`?

ES6 doesn't look proper category. Though it has not done, https://github.com/eslint/eslint/issues/7991 has been accepted.